exonum/src/api/private/system.rs

```diff
@@ -156,31 +155,16 @@ impl SystemApi {
         self.node_channel.send_external_message(message)?;
         Ok(())
     }
-
-    fn peer_add(&self, ip_str: &str) -> Result<(), ApiError> {
-        let addr: SocketAddr = ip_str.parse()?;
-        self.node_channel.peer_add(addr)?;
-        Ok(())
-    }
 }
 
 impl Api for SystemApi {
     fn wire(&self, router: &mut Router) {
 
         let self_ = self.clone();
         let peer_add = move |req: &mut Request| -> IronResult<Response> {
-            let map = req.get_ref::<Params>().unwrap();
-            match map.find(&["ip"]) {
-                Some(&ParamsValue::String(ref ip_str)) => {
-                    self_.peer_add(ip_str)?;
-                    self_.ok_response(&::serde_json::to_value("Ok").unwrap())
-                }
-                _ => {
-                    Err(ApiError::IncorrectRequest(
-                        "Required parameter of peer 'ip' is missing".into(),
-                    ))?
-                }
-            }
+            let addr: SocketAddr = required_param(req, "ip")?;
+            self_.node_channel.peer_add(addr).map_err(ApiError::from)?;
+            self_.ok_response(&::serde_json::to_value("Ok").unwrap())
```

Comment:
What about to move a `::serde_json::to_value(&str).unwrap()` into `Api::ok_response` method ?

Reply:
Let's do it in separate PR

Reply:
Maybe, it makes sense to use [`json!` macro](https://docs.serde.rs/serde_json/macro.json.html), like `json!("Ok")`?